Common/Data/DividendYieldProvider.cs

```diff
@@ -95,8 +93,13 @@ public decimal GetDividendYield(DateTime date)
         /// </summary>
         protected void LoadDividendYieldProvider()
         {
-            var directory = Path.Combine(Globals.DataFolder, "equity", "usa", "factor_files", $"{_symbol.Value.ToLowerInvariant()}.csv");
-            _dividendYieldRateProvider = FromCsvFile(directory);
+            var factorFileProvider = Composer.Instance.GetPart<IFactorFileProvider>();
+            var corporateFactors = factorFileProvider
+                .Get(_symbol)
+                .Select(factorRow => factorRow as CorporateFactorRow)
+                .Where(corporateFactor => corporateFactor != null);
+
+            _dividendYieldRateProvider = FromCorporateFactorRow(corporateFactors);
 
             _lastDividendYieldDate = DateTime.UtcNow.Date;
```

Comment:
I don't think need to forward the last dividend yield until now, can actually just keep the last dividend yield value on the actual last date we have data for, and return it in `GetDividendYield(DateTime date)` instead of `DividendYieldRateProvider[_lastDividendYieldDate]`